### ISSUE TITLE
torch YatConvTranspose{1,2,3}D crash for groups > 1 — wrong ones_kernel first-dim size

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#22]
## Summary

All three torch YAT transposed convolution layers (`YatConvTranspose1D`, `YatConvTranspose2D`, `YatConvTranspose3D`) crash when constructed with `groups > 1`. The `ones_kernel` used to compute `||input_patches||²` is built with the **standard-conv** layout (`in_channels // groups, out_channels // groups, *k`) instead of the **transposed-conv** layout that PyTorch actually expects (`in_channels, out_channels // groups, *k`).

## Affected lines

Same pattern in all three files:

`src/nmn/torch/layers/yat_conv_transpose1d.py:212-215`
`src/nmn/torch/layers/yat_conv_transpose2d.py:212-215`
`src/nmn/torch/layers/yat_conv_transpose3d.py:212-215`

```python
in_channels_per_group = self.in_channels // self.groups
out_channels_per_group = self.out_channels // self.groups
ones_kernel_shape = (in_channels_per_group, out_channels_per_group) + self.kernel_size
ones_kernel = torch.ones(ones_kernel_shape, device=input.device, dtype=input.dtype)

patch_sq_sum_map = F.conv_transpose1d(
    input_squared, ones_kernel, None, self.stride, self.padding,
    output_padding, self.groups, self.dilation
)
```

PyTorch's `F.conv_transpose*d` weight layout is `(in_channels, out_channels // groups, *kernel_size)`. With `in_channels=8, out_channels=16, groups=2`, the ones_kernel is built as `(4, 8, k)` but PyTorch expects `(8, 8, k)` — so the conv call rejects the input as having "8 channels but weight expects 4".

The first dim should always be `in_channels`, not `in_channels // groups`. The actual learnable `self.weight` is already (correctly) `(in_channels, out_channels // groups, *k)` because it comes from `nn.ConvTranspose*d.__init__`.

## Repro

```python
import torch
from nmn.torch.layers.yat_conv_transpose1d import YatConvTranspose1D
from nmn.torch.layers.yat_conv_transpose2d import YatConvTranspose2D
from nmn.torch.layers.yat_conv_transpose3d import YatConvTranspose3D

for cls, x_shape in [
    (YatConvTranspose1D, (2, 8, 10)),
    (YatConvTranspose2D, (2, 8, 10, 10)),
    (YatConvTranspose3D, (2, 8, 6, 6, 6)),
]:
    for g in [1, 2, 4]:
        try:
            cls(8, 16, 3, groups=g)(torch.randn(*x_shape))
            print(f"{cls.__name__} groups={g}: OK")
        except RuntimeError as e:
            print(f"{cls.__name__} groups={g}: {str(e)[:80]}")
```

```
YatConvTranspose1D groups=1: OK
YatConvTranspose1D groups=2: Given transposed=1, weight of size [4, 8, 3], expected input[2, 8, 10] to have 4 channels, but got 8 channels instead
YatConvTranspose1D groups=4: Given groups=4, expected weight to be at least 4 at dimension 0, but got weight of size [2, 4, 3] instead
... (same for 2D and 3D)
```

## Suggested fix

```python
in_channels_per_group = self.in_channels // self.groups
out_channels_per_group = self.out_channels // self.groups
ones_kernel_shape = (self.in_channels, out_channels_per_group) + self.kernel_size
```

(Drop `in_channels_per_group` from the computation since it's only used to (mis)build the kernel. The summed result will still be `(B, out_channels, ...)` because conv_transpose-with-ones over groups sums each group's input channels into its own group's output channels.)

Once the kernel is the right shape, the subsequent `patch_sq_sum_map` will broadcast cleanly against `dot_prod_map` (shape `(B, out_channels, ...)`), and the `view_shape = (1, -1, 1, ...)` reshape of `kernel_sq_sum_per_filter` already produces `(1, out_channels // groups, 1, ...)` — which broadcasts correctly against the full `out_channels` map since within each of the `groups` groups, the filters of that group share the same in-channel partition.

A regression test should cover at least `groups in {2, 4, in_channels}` (depthwise) for all three transposed variants.
